src/openforms/forms/tests/public_api/test_endpoints.py

```diff
@@ -0,0 +1,78 @@
+from django.contrib.auth.models import Permission
+
+from rest_framework import status
+from rest_framework.reverse import reverse
+from rest_framework.test import APITestCase
+
+from openforms.accounts.tests.factories import TokenFactory, UserFactory
+
+from ..factories import CategoryFactory, FormFactory
+
+
+class TestPublicFormEndpoint(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        user = UserFactory.create()
+        user.user_permissions.add(Permission.objects.get(codename="view_form"))
+
+        blue_cat = CategoryFactory.create(name="Blue")
+        red_cat = CategoryFactory.create(name="Red")
+
+        FormFactory.create(slug="deleted-form", deleted_=True)
+        FormFactory.create(slug="1-form", category=blue_cat)
+        FormFactory.create(slug="2-form", category=blue_cat)
+        FormFactory.create(slug="3-form", category=red_cat)
+
+        cls.blue_cat = blue_cat
+        cls.user = user
+
+    def test_cant_access_without_token(self):
+        response = self.client.get(
+            reverse(
+                "api:public:forms:forms-list",
+            )
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_cant_access_without_view_form_permission(self):
+        user = UserFactory.create()
+        token = TokenFactory(user=user)
```

Comment:
```suggestion
        token = TokenFactory()
```
it will create a user automatically